skcosmo/feature_selection/voronoi_fps.py

```diff
@@ -0,0 +1,229 @@
+
+    This implicit tessellation enabled a more efficient evaluation of the FPS --
+    at each iteration, we need only consider for selection those points at the
+    boundaries of the Voronoi polyhedra, and when updating the tessellation
+    we need only consider moving those points whose Hausdorff distance is
+    greater than half of the distance between the corresponding Voronoi center
+    and the newly selected point, per the triangle equality.
+
+    Parameters
+    ----------
+
+    n_features_to_select: int or float, default=None
+        The number of features to select. If `None`,
+        half of the features are selected. If integer, the parameter is the
+        absolute number of features to select. If float between 0 and 1, it is
+        the fraction of features to select.
+
+    initialize: int or 'random', default=0
+        Index of the first feature to be selected. If 'random', picks a random
+        value when fit starts.
+
+    tolerance: float, defaults to 1E-12
+        threshold below which distances will be considered 0
+
+    progress_bar: boolean, default=False
+        option to use `tqdm <https://tqdm.github.io/>`_
+        progress bar to monitor selections
+
+    n_trial_calculation: integer, default=4
+        Number of calculations used for the switching point between Voronoi FPS
+        and traditional FPS (for detail look at full_fraction).
+
+    full_fraction: float, default=None
+        Proportion of calculated distances from the total number of features at
+        which the switch from Voronoi FPS to FPS occurs.
+        At a certain number of distances to be calculated,
+        the use of Voronoi FPS becomes unreasonably expensive due to the associated
+        costs connected with reading data from the memory. The switching point
+        depends on many conditions, and it is determined "in situ" for optimal
+        use of the algorithm. Determination is done with a few test calculations
+        and memory operations.
+    """
+
+    def __init__(
+        self,
+        n_features_to_select=None,
+        initialize=0,
+        tolerance=1e-12,
+        progress_bar=False,
+        n_trial_calculation=4,
+        full_fraction=None,
+    ):
+
+        self.n_trial_calculation_ = n_trial_calculation
+        self.full_fraction = full_fraction
+
+        super().__init__(
+            n_features_to_select=n_features_to_select,
+            progress_bar=progress_bar,
+            initialize=initialize,
+            tolerance=tolerance,
+        )
+
+    def _init_greedy_search(self, X, y, n_to_select):
+        """Initializes the search. Prepares an array to store the selected
+        features. This function also determines the switching point if it was not
+        given during initialization."""
+
+        n_features = X.shape[1]
+        """index of the voronoi cell associated with each of the columns of X"""
+        self.vlocation_of_idx = np.full(n_features, -1)
```

Comment:
I think this needs to be reversed to be interpreted as a docstring:

```suggestion
        self.vlocation_of_idx = np.full(n_features, -1)
        """index of the voronoi cell associated with each of the columns of X"""
```

and similarly for the other attributes below